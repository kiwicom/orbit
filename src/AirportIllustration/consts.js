// @flow

// eslint-disable-next-line import/prefer-default-export
export const NAMES = [
  "BGYFastTrack",
  "BUDFastTrack",
  "MRSSmartPass",
  "NCEFastTrack",
  "PRGSmartPass",
  "VCESmartPass",
];
