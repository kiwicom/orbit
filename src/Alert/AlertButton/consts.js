// @flow
export const TYPE_OPTIONS = {
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
  INFO_SUBTLE: "infoSubtle",
  SUCCESS_SUBTLE: "successSubtle",
  WARNING_SUBTLE: "warningSubtle",
  CRITICAL_SUBTLE: "criticalSubtle",
};

export const TOKENS = {
  backgroundButton: "backgroundButton",
  backgroundButtonHover: "backgroundButtonHover",
  backgroundButtonActive: "backgroundButtonActive",
  backgroundButtonFocus: "backgroundButtonFocus",
  colorTextButton: "colorTextButton",
  borderColorButtonFocus: "borderColorButtonFocus",
};
