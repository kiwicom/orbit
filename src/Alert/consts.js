// @flow
export const TYPE_OPTIONS = {
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
};

export const TOKENS = {
  backgroundAlert: "backgroundAlert",
  colorIconAlert: "colorIconAlert",
  colorTextAlert: "colorTextAlert",
  colorTextLinkAlertHover: "colorTextLinkAlertHover",
  colorTextLinkAlertFocus: "colorTextLinkAlertFocus",
  colorBorderAlert: "colorBorderAlert",
};

export const CLOSE_BUTTON_DATA_TEST = "AlertCloseButton";
