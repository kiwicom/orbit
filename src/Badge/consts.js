// @flow
export const TYPE_OPTIONS = {
  NEUTRAL: "neutral",
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
  DARK: "dark",
  WHITE: "white",
  INFO_INVERTED: "infoInverted",
  CRITICAL_INVERTED: "criticalInverted",
};

export const TOKENS = {
  background: "background",
  color: "color",
};
