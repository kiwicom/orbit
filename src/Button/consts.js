// @flow
export const TYPE_OPTIONS = {
  PRIMARY: "primary",
  SECONDARY: "secondary",
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
  FACEBOOK: "facebook",
  GOOGLE: "google",
  WHITE: "white",
};

export const SIZE_OPTIONS = {
  SMALL: "small",
  NORMAL: "normal",
  LARGE: "large",
};

export const TOKENS = {
  // Size tokens
  heightButton: "heightButton",
  loadingWidth: "loadingWidth",
  loadingHeight: "loadingHeight",
  fontSizeButton: "fontSizeButton",
  paddingButton: "paddingButton",
  paddingButtonWithIcons: "paddingButtonWithIcons",
  paddingButtonWithLeftIcon: "paddingButtonWithLeftIcon",
  paddingButtonWithRightIcon: "paddingButtonWithRightIcon",
  marginRightIcon: "marginRightIcon",
  // Type tokens
  backgroundButton: "backgroundButton",
  backgroundButtonHover: "backgroundButtonHover",
  backgroundButtonActive: "backgroundButtonActive",
  backgroundButtonBordered: "backgroundButtonBordered",
  backgroundButtonBorderedHover: "backgroundButtonBorderedHover",
  backgroundButtonBorderedActive: "backgroundButtonBorderedActive",
  colorTextButton: "colorTextButton",
  colorTextButtonBordered: "colorTextButtonBordered",
  colorTextButtonHover: "colorTextButtonHover",
  colorTextButtonBorderedHover: "colorTextButtonBorderedHover",
  colorTextButtonActive: "colorTextButtonActive",
  colorTextButtonBorderedActive: "colorTextButtonBorderedActive",
  borderColorButton: "borderColorButton",
  borderColorButtonHover: "borderColorButtonHover",
  borderColorButtonActive: "borderColorButtonActive",
};

export const BUTTON_STATES = {
  DEFAULT: "default",
  HOVER: "hover",
  ACTIVE: "active",
  FOCUS: "focus",
};
