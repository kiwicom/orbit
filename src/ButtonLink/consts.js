// @flow
export const TYPES = {
  PRIMARY: "primary",
  SECONDARY: "secondary",
};

export const SIZES = {
  SMALL: "small",
  NORMAL: "normal",
  LARGE: "large",
};

export const TOKENS = {
  backgroundButton: "backgroundButton",
  backgroundButtonHover: "backgroundButtonHover",
  backgroundButtonActive: "backgroundButtonActive",
  colorTextButton: "colorTextButton",
  colorTextButtonHover: "colorTextButtonHover",
  colorTextButtonActive: "colorTextButtonActive",
  heightButton: "heightButton",
  fontSizeButton: "fontSizeButton",
  paddingButton: "paddingButton",
  paddingButtonWithIcons: "paddingButtonWithIcons",
  paddingButtonWithLeftIcon: "paddingButtonWithLeftIcon",
  paddingButtonWithRightIcon: "paddingButtonWithRightIcon",
  marginRightIcon: "marginRightIcon",
};

export const BUTTON_STATES = {
  ACTIVE: "active",
  FOCUS: "focus",
};
