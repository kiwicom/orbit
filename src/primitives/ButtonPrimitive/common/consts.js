// @flow
export const SIZE_OPTIONS = {
  SMALL: "small",
  NORMAL: "normal",
  LARGE: "large",
};

export const TOKENS = {
  heightButton: "heightButton",
  fontSizeButton: "fontSizeButton",
  paddingButton: "paddingButton",
  paddingButtonWithIcons: "paddingButtonWithIcons",
  paddingButtonWithLeftIcon: "paddingButtonWithLeftIcon",
  paddingButtonWithRightIcon: "paddingButtonWithRightIcon",
  spinnerWidth: "spinnerWidth",
  spinnerHeight: "spinnerHeight",
  marginRightIcon: "marginRightIcon",
};

export const BUTTON_STATES = {
  DEFAULT: "default",
  HOVER: "hover",
  ACTIVE: "active",
  FOCUS: "focus",
};
