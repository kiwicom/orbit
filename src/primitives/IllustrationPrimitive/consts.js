// @flow

export const SIZE_OPTIONS = {
  SMALL: "small",
  MEDIUM: "medium",
  LARGE: "large",
  DISPLAY: "display",
};

export const baseURL = "//images.kiwi.com";
