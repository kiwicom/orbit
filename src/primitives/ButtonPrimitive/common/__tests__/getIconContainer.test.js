// @flow
import getIconContainer from "../getIconContainer";
import theme from "../../../../defaultTheme";
import { SIZE_OPTIONS } from "../consts";

const iconForeground = {
  foreground: null,
  foregroundHover: null,
  foregroundActive: null,
};

describe("ButtonPrimitive: getIconContainer function", () => {
  it("should match when only icon", () => {
    expect(getIconContainer({ iconLeft: true, theme, iconForeground })).toMatchSnapshot();
  });
  it("should match snapshot", () => {
    Object.keys(SIZE_OPTIONS).forEach(key => {
      expect(
        getIconContainer({
          size: SIZE_OPTIONS[key],
          children: "Lorem ipsum",
          theme,
          iconForeground,
        }),
      ).toMatchSnapshot();
    });
  });
});
