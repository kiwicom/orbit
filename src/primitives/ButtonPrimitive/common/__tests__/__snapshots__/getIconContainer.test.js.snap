// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimitive: getIconContainer function should match snapshot 1`] = `
Object {
  "icons": Object {
    "foreground": null,
    "foregroundActive": null,
    "foregroundHover": null,
    "height": "16px",
    "leftMargin": "0 8px 0 0",
    "rightMargin": "0 0 0 8px",
    "width": "16px",
  },
}
`;

exports[`ButtonPrimitive: getIconContainer function should match snapshot 2`] = `
Object {
  "icons": Object {
    "foreground": null,
    "foregroundActive": null,
    "foregroundHover": null,
    "height": "24px",
    "leftMargin": "0 8px 0 0",
    "rightMargin": "0 0 0 8px",
    "width": "24px",
  },
}
`;

exports[`ButtonPrimitive: getIconContainer function should match snapshot 3`] = `
Object {
  "icons": Object {
    "foreground": null,
    "foregroundActive": null,
    "foregroundHover": null,
    "height": "24px",
    "leftMargin": "0 12px 0 0",
    "rightMargin": "0 0 0 12px",
    "width": "24px",
  },
}
`;

exports[`ButtonPrimitive: getIconContainer function should match when only icon 1`] = `
Object {
  "icons": Object {
    "foreground": null,
    "foregroundActive": null,
    "foregroundHover": null,
    "height": "24px",
    "width": "24px",
  },
}
`;
